src/server.rs

```diff
@@ -0,0 +1,213 @@
+use std::sync::Arc;
+
+use alloy::eips::{BlockId, BlockNumberOrTag};
+use alloy::primitives::{Address, B256};
+use alloy::rpc::types::serde_helpers::JsonStorageKey;
+use alloy_primitives::U64;
+use alloy_rpc_types_engine::{
+    ExecutionPayloadV3, ForkchoiceState, ForkchoiceUpdated, PayloadAttributes, PayloadId,
+    PayloadStatus,
+};
+use alloy_rpc_types_eth::{Block, EIP1186AccountProofResponse};
+use jsonrpsee::core::{async_trait, ClientError, RpcResult};
+use jsonrpsee::http_client::HttpClient;
+use jsonrpsee::proc_macros::rpc;
+use jsonrpsee::types::ErrorCode;
+use tracing::error;
+
+#[rpc(server, client, namespace = "eth")]
+pub trait EthApi {
+    /// Returns the chain ID of the current network.
+    #[method(name = "chainId")]
+    async fn chain_id(&self) -> RpcResult<Option<U64>>;
+
+    /// Returns information about a block by number.
+    #[method(name = "getBlockByNumber")]
+    async fn block_by_number(
+        &self,
+        number: BlockNumberOrTag,
+        full: bool,
+    ) -> RpcResult<Option<Block>>;
+
+    /// Returns information about a block by hash.
+    #[method(name = "getBlockByHash")]
+    async fn block_by_hash(&self, hash: B256, full: bool) -> RpcResult<Option<Block>>;
+
+    /// Returns the account and storage values of the specified account including the Merkle-proof.
+    /// This call can be used to verify that the data you are pulling from is not tampered with.
+    #[method(name = "getProof")]
+    async fn get_proof(
+        &self,
+        address: Address,
+        keys: Vec<JsonStorageKey>,
+        block_number: Option<BlockId>,
+    ) -> RpcResult<EIP1186AccountProofResponse>;
+}
+
+#[rpc(server, client, namespace = "engine")]
+pub trait EngineApi {
```

Comment:
Can reth traits be used?

Reply:
This is a subset of the Reth API right? If we import the whole Eth API we have to implement all the methods?